bitnami/elasticsearch/templates/master/statefulset.yaml

```diff
@@ -391,6 +391,11 @@ spec:
           persistentVolumeClaim:
             claimName: {{ .Values.master.persistence.existingClaim }}
   {{- else }}
+  {{- if .Values.persistentVolumeClaimRetentionPolicy.enabled }}
+          persistentVolumeClaimRetentionPolicy:
+            whenDeleted: {{ .Values.persistentVolumeClaimRetentionPolicy.whenDeleted }}
+            whenScaled: {{ .Values.persistentVolumeClaimRetentionPolicy.whenScaled }}
+  {{- end }}
```

Comment:
You forgot the `master/data` in the values, it should be `.Values.<component>`

```yaml
  {{- if .Values.master.persistentVolumeClaimRetentionPolicy.enabled }}
          persistentVolumeClaimRetentionPolicy:
            whenDeleted: {{ .Values.master.persistentVolumeClaimRetentionPolicy.whenDeleted }}
            whenScaled: {{ .Values.master.persistentVolumeClaimRetentionPolicy.whenScaled }}
  {{- end }}
```

Reply:
`helm install elasticsearch bitnami/elasticsearch --dry-run` is working now - thanks!